pygfx/renderers/wgpu/_pipeline.py

```diff
@@ -607,27 +654,71 @@ def update_vertex_buffer_descriptors(self):
                 ],
             }
             vertex_buffer_descriptors.append(vbo_des)
+
+        instance_buffer = self.resources.get("instance_buffers", None)
+        if instance_buffer is not None:
```

Comment:
I'm not a fan of this approach, to be honest. Using a storage buffer for the instance matrices (as `main` currently does) can avoid quite a bit of logic and make the codebase simpler.